packages/strapi-helper-plugin/lib/src/utils/request.js

```diff
@@ -82,11 +82,20 @@ function serverRestartWatcher(response) {
    // Set headers
    options.headers = Object.assign({
      'Content-Type': 'application/json',
+     'X-Forwarded-Host': 'strapi',
    }, options.headers);
 
+   const token = auth.getToken();
+
+   if (token) {
+     options.headers = Object.assign({
```

Comment:
The order of the assigned objects is not correct. Also, you don't need to reassign the options.headers value because it mutates the first object.

```js
Object.assign(options.headers, {
  'Authorization': `Bearer ${token}`
});
```